### DISCUSSION TITLE
Would runtime configurations be a good idea?

### DISCUSSION THREAD

**Discussion by dkbast** [serverpod/serverpod#2144] [Q&A]
I'm thinking about a scenario where I want to distribute my project as a binary to be hosted by my customers - they should be able to configure publicHost, database etc pp themselves, so that they can make use of their own infrastructure. 

The same goes for publishing a prebuilt Docker image of my project - I would like to configure everything at runtime to make it possible for the project to coexist with others.

That would probably also be valuable when running serverpod in a larger cloud environment?

**Comment by Isakdl** [comment]
Yes absolutely this is a needed feature, generally, I think we need to support many different ways to inject the environment variables with some defined order of precedence so you can override values, etc.
